aten/src/ATen/Dispatch.h

```diff
@@ -123,6 +123,38 @@ inline void deprecated_AT_DISPATCH_ALL_TYPES_AND_HALF_AND_COMPLEX() {}
     }                                                                        \
   }()
 
+ #define AT_DISPATCH_FLOATING_TYPES_AND_COMPLEX(TYPE, NAME, ...)                                                  \
```

Comment:
It would be good to get some explanation of what happened to the dispatch macros here. They've historically been pretty messy and we've been trying to clean them up. In interest of not regressing here, it would be good to understand what changes we're applying here.